enterprise-metrics/main.libsonnet

```diff
@@ -0,0 +1,379 @@
+local cortex =
```

Comment:
would anyone ever want to override this `cortex`? If so, would it be better as a hidden?

Reply:
That's a really good question and one I have been struggling with a lot through the course of writing this library.

While `cortex` is a local variable, I can completely control the user interface. For instance, all modules are represented by a consistent `{ args, container, [job || deployment || statefulSet], service }` where by their nature, hidden fields function as a more flexible function input and the visible fields become output. Sure, a user can do something nonsensical like `{ args:: 'this is now a string' }` and it would break the functioning of the rest of that module but I would imagine that it would be fairly easy for a user to understand the resulting error.

If I make`cortex` a hidden field, I then expose the entire `cortex` interface in my library which gives users great flexibility but at a huge overhead in complexity to debug. Admittedly the Jsonnet errors are getting better so that this isn't quite so difficult to decipher any more.

Right now I'm leaning towards keeping this private and reserving the option to expose `cortex` in the future.